src/interpreter.rs

```diff
@@ -18,9 +20,20 @@ use sota::Sota;
 pub trait Interpreter<I, O> {
     fn interpret(&mut self, input: I, otx: &Sender<O>);
 
-    fn run(&mut self, irx: Receiver<I>, otx: Sender<O>) {
+    fn run(&mut self, irx: Receiver<I>, otx: Sender<O>, wg: WaitGroup) {
+        let cooldown = Duration::from_millis(100);
+
         loop {
-            self.interpret(irx.recv().expect("interpreter sender closed"), &otx);
+            let input   = irx.recv().expect("interpreter sender closed");
+            let started = time::precise_time_ns();
+
+            wg.add(1);
+            debug!("interpreter starting: {}", started);
+            self.interpret(input, &otx);
+
+            thread::sleep(cooldown); // let any further work commence
```

Comment:
@stevana: not a big fan of this, but it forces any new counter increments before this decrement